nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -176,10 +184,33 @@ in
 
     systemd.user.services.dbus.wantedBy = [ "default.target" ];
 
-    programs.dconf.profiles.gdm = pkgs.writeText "dconf-gdm-profile" ''
-      system-db:local
-      ${gdm}/share/dconf/profile/gdm
-    '';
+    programs.dconf.profiles.gdm =
+    let
+      customDconf = pkgs.writeTextFile {
+        name = "gdm-dconf";
+        destination = "/dconf/gdm-custom";
+        text = ''
+          ${optionalString (!cfg.gdm.autoSuspend) ''
+            [org/gnome/settings-daemon/plugins/power]
+            sleep-inactive-ac-timeout=0
+            sleep-inactive-battery-timeout=0
+          ''}
+        '';
+      };
+
+      customDconfDb = pkgs.stdenv.mkDerivation {
+        name = "gdm-dconf-db";
+        buildCommand = ''
+          ${pkgs.gnome3.dconf}/bin/dconf compile $out ${customDconf}/dconf
+        '';
+      };
+    in pkgs.stdenv.mkDerivation {
+      name = "dconf-gdm-profile";
+      buildCommand = ''
+        echo "file-db:${customDconfDb}" > $out
```

Comment:
By having `file-db` as the first line in a dconf profile it makes it read only. Is that intentional?

> https://gitlab.gnome.org/GNOME/dconf/blob/master/engine/dconf-engine-profile.c#L80

Reply:
I think it's a good thing, a system user like GDM's settings should be specified in Nix, not with the dconf tool or whatever.

Reply:
Currently there's no real way to do this in Nix though so it would be best to do what upstream has

Reply:
Doesn't this commit add it, at least for gdm? We could add an `extraDconf` option.

Reply:
An ad-hoc solution just for `gdm` isn't really desired, we'd want https://github.com/NixOS/nixpkgs/issues/54150 for that.

Reply:
OK, but since that doesn't exist yet, why don't we add this first and then switch it to that solution behind the scenes when it's ready? Besides, I don't know if mutable databases even work given that gdm doesn't have a home directory and /etc is read-only.

Reply:
gdm does have a home directory https://github.com/NixOS/nixpkgs/blob/c6748ef63f2e478b89d56023ab5f232b26125e21/nixos/modules/services/x11/display-managers/gdm.nix#L120 and the permissions for `/etc` on my NixOS install are writable
```
drwxr-xr-x - root root  1 Jul 14:43 etc
```

Reply:
OK, I made it put the custom DB behind `user-db`.